app/views/shared/_navigation_bar.html.erb

```diff
@@ -15,6 +15,10 @@
     <a class="inline-block border rounded py-1 px-3" href="/?domain=UpgradeJS.com">UpgradeJS Links</a>
   </li>
 
+  <li class="mr-3">
+    <%= button_to "Execute 'Read' task", execute_read_task_path, class: "inline-block border rounded py-1 px-3", method: :post %>
```

Comment:
How about we use the `disable_with` data option to disable the button after it is clicked?